### PR TITLE
Fix Make targets

### PR DESCRIPTION
**Pull request by tamird**
This PR fixes the make targets by:
- Updating `protoc-gen-go/testdata regenerate`
- Making the `sed` invocation BSD `sed` friendly

It also regenerates the descriptor and plugin protos from google/protobuf at v3.0.0-alpha-3.1.

### CODE REVIEW

**Comment by dsymonds** [comment]
Thanks, but I can't see the need for any of this. I regenerate some things here when needed, but nothing new in descriptor.proto is needed. This is just churn.

**Comment by tamird** [comment]
I'm happy to remove the last commit if you think it's unneeded. The first two commits are the minimum to get `make` working on systems that have non-gnu sed.

**Comment by dsymonds** [comment]
Okay, let's do just that. Squash the commits please.

**Comment by tamird** [comment]
Done

File: Makefile [hunk 1]
```diff
@@ -51,6 +51,6 @@ nuke:
 regenerate:
 	make -C protoc-gen-go/descriptor regenerate
 	make -C protoc-gen-go/plugin regenerate
-	make -C protoc-gen-go/testdata regenerate
+	make -C protoc-gen-go/testdata buildprotos
```
Comment (dsymonds) [inline Makefile]: actually, I don't  know why that target is named differently. Can you revert this change and fix `protoc-gen-go/testdata/Makefile` instead?

Comment (tamird) [inline Makefile]: sure, done.

**Comment by dsymonds** [comment]
LGTM

**Comment by dsymonds** [comment]
Merged in 09bcc32. Thanks.
